cub/cub/block/block_load.cuh

```diff
@@ -879,12 +890,16 @@ class BlockLoad
     {}
 
     // attempts vectorization (pointer)
-    template <typename>
     _CCCL_DEVICE _CCCL_FORCEINLINE void Load(const T* block_ptr, T (&dst_items)[ITEMS_PER_THREAD])
     {
       InternalLoadDirectBlockedVectorized<LOAD_DEFAULT>(linear_tid, block_ptr, dst_items);
     }
 
+    _CCCL_DEVICE _CCCL_FORCEINLINE void Load(T* block_ptr, T (&dst_items)[ITEMS_PER_THREAD])
```

Comment:
```suggestion
    _CCCL_DEVICE _CCCL_FORCEINLINE void Load(const T* block_ptr, T (&dst_items)[ITEMS_PER_THREAD])
```

Reply:
We already have an existing `const T*` overload declared. However that overload isn't chosen if the user gives us a pointer-to-non-const, hence we want to add this overload as is (see number 2 in [this comment](https://github.com/NVIDIA/cccl/pull/3517#issuecomment-2612035631)).

Reply:
Alright. Could you please add a comment about this to the new non-const overload? Because I will forget about that oddity and refactor this at some point again, if a comment does not prevent me :D Thx!